src/components/views/LocalModList/SearchAndSort.vue

```diff
@@ -0,0 +1,132 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { DeferredInput } from '../../all';
+import GameManager from '../../../model/game/GameManager';
+import { SortDirection } from '../../../model/real_enums/sort/SortDirection';
+import { SortLocalDisabledMods } from '../../../model/real_enums/sort/SortLocalDisabledMods';
+import { SortNaming } from '../../../model/real_enums/sort/SortNaming';
+import ManagerSettings from '../../../r2mm/manager/ManagerSettings';
+
+@Component({
+    components: {
+        DeferredInput,
+    }
+})
+export default class SearchAndSort extends Vue {
+    settings = new ManagerSettings();
+
+    get order() {
+        return this.$store.state.profile.order;
+    }
+
+    set order(value: SortNaming) {
+        this.$store.commit('profile/setOrder', value);
+        this.settings.setInstalledSortBy(value);
+    }
+
+    get direction() {
+        return this.$store.state.profile.direction;
+    }
+
+    set direction(value: SortDirection) {
+        this.$store.commit('profile/setDirection', value);
+        this.settings.setInstalledSortDirection(value);
```

Comment:
Likewise here, shouldn't the store be the primary "API" for making these changes, and as such take care of syncing the state to where it should get synced?

Alternatively, the settings could subscribe itself to track the profile store I suppose.

Reply:
As per Discord discussions, to make this change properly we'd need to also have the active game and profile available in Vuex. That and these changes are now on my todo list but meanwhile I won't block these changes.